cardano-cli/src/Cardano/CLI/Legacy/Run/Transaction.hs

```diff
@@ -146,10 +149,19 @@ runLegacyTransactionBuildRawCmd
       & hoistEither
       & firstExceptT TxCmdTxUpdateProposalValidationError
 
+  upperBound <-
+    caseByronOrShelleyBasedEra
+      (\w -> case mUpperBound of
+        Nothing -> pure $ TxValidityNoUpperBound w
+        Just _ -> left $ TxCmdTxValidityUpperBoundValidationError $ TxValidityUpperBoundNotSupported anyEra
+      )
+      (\w -> pure $ TxValidityUpperBound w mUpperBound)
+      era
```

Comment:
For the legacy commands we require validation so we do it here.  If the era does not support a value, and a value was parsed by the CLI we raise an error.